sig-instrumentation/annual-report-2022.md

```diff
@@ -0,0 +1,150 @@
+   - Beta
+     - [1602 - Structured Logging](https://git.k8s.io/community/$link/README.md) - 1.23
+     - [1748 - Pod resource requests/limits metrics](https://git.k8s.io/community/$link/README.md) - 1.22
+   - Alpha
+     - [2305 - Metrics Cardinality Enforcement](https://git.k8s.io/community/$link/README.md) - 1.21
+     - [647 - API Server Tracing](https://git.k8s.io/community/$link/README.md) - 1.22
+     - [2845 - Deprecate klog-specific flags in k8s components](https://git.k8s.io/community/$link/README.md) - 1.23
+   - Pre-alpha
+     - [2831 - Kubelet OpenTelemetry Tracing](https://git.k8s.io/community/$link/README.md) - alpha in 1.24
+
+## Project health
+
+1. What areas and/or subprojects does your group need the most help with?
+   Any areas with 2 or fewer OWNERs? (link to more details)
+
+   - [kubernetes-sigs/custom-metrics-apiserver](https://github.com/kubernetes-sigs/custom-metrics-apiserver/blob/master/OWNERS) has 1 active approver
+   - [kubernetes-sigs/metrics-server](https://github.com/kubernetes-sigs/metrics-server/blob/master/OWNERS) has 1 active approver
+   - [kubernetes-sigs/prometheus-adapter](https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/OWNERS_ALIASES) has 1 active approver
+
+2. What metrics/community health stats does your group care about and/or measure?
+
+   - Devstats
+     - [Review Load](https://k8s.devstats.cncf.io/d/80/pr-workload-per-sig-and-repository-chart?orgId=1&var-sigs=%22instrumentation%22&var-repo_name=kubernetes%2Fkubernetes&var-repo=kuberneteskubernetes&from=now-1y&to=now) has been relatively level over the year
+     - [Time to Approve and Merge](https://k8s.devstats.cncf.io/d/44/pr-time-to-approve-and-merge?orgId=1&var-period=d7&var-repogroup_name=SIG%20Instrumentation&var-repo_name=kubernetes%2Fkubernetes&var-apichange=All&var-size_name=All&var-kind_name=All) has been relatively low over the last year.
+   - Meeting attendance
+       - Meeting attendance is ~10 each week
+       - Triage attendance is ~5 each week
+   - Enhancement velocity
+       - 7 Active enhancements over the year
+
+3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
+   to activities or programs that provide useful context or allow easy participation?
+
+   -
+
+4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
+   does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?
+
+   - N/A, we don't have special training or requirements
+
+5. Does the group have contributors from multiple companies/affiliations?
+
+   - Yes: Red Hat, Google, Intel, ...
+
+6. Are there ways end users/companies can contribute that they currently are not?
+   If one of those ways is more full time support, what would they work on and why?
+
+   - Yes
+   -
```

Comment:
```suggestion
SIG leads performed a [staffing and gap analysis](https://docs.google.com/document/d/1qeoP6i7GBTVJuJE1AGY5iU9dqmAOxrjqkfNQ2-rBeyI/edit#heading=h.849b7ydpl7ip) for the SIG's projects. We definitely can use more help, and need more part-time/full-time contributors.

### Summary

- **KTLO:** 4 leads at 5%, 2 maintainers at 10% for core, 9-10 maintainers at 5% for 5 subprojects (can have overlap between roles, but need a minimum of 3-4 part-time contributors at 25%)
- **Feature work:** needs significant ongoing additional investment, minimum of 2-3 FT devs or features will continue to slip

### Details

- **KTLO:** Requires two experienced part-time maintainers at 10% (e.g. 2x4h = 8h/wk) in addition to the SIG leadership (chairs/TLs @ minimum of 2h/wk)
  - Chairs/TLs currently perform the bulk of this work but even amongst the four of them, **do not** have 8h total weekly allocated
- **Feature work:** requires significantly more investment from development and review time. 
  - E.g. Structured logging initiative requires a minimum of 2 FT staff for the duration of feature development from beta -> GA
  - Additional metric stability classes and graduation: 1PT maintainer at 25% for the duration of feature development
  - SIG currently owns a number of KEPs stuck in alpha/beta due to lack of dev resources:
      - https://github.com/kubernetes/enhancements/issues/383 
      - https://github.com/kubernetes/enhancements/issues/1668
      - https://github.com/kubernetes/enhancements/issues/1753 
      - https://github.com/kubernetes/enhancements/issues/2305 

### Subprojects

- Subprojects are currently mature/stable and mainly have KTLO needs
- kube-state-metrics: 2-3 experienced maintainers at 5%, currently staffed
- Metrics-server: 2 experienced maintainers at 5%, currently staffed
- Custom-metrics-apiserver: 1 experienced maintainer at 5%, currently staffed
- Klog: 2 experienced maintainers at 5%, has **no current staffing**
- Prometheus-adapter: 2 experienced maintainers at 5%, has **no current staffing**
```

Reply:
The number of maintainers listed here for subprojects differs from what I wrote above, and I believe this version is correct (devstats didn't have anything for s-urbaniak, but it looks like he has done some reviews recently).  I commented above to fix.

Reply:
This version wasn't based on devstat but rather on each project maintainer's feelings about the health of the project. Although some of them have been partially staffed this year (e.g prometheus-adapter), none of the maintainers have currently time to invest in it.